### PR TITLE
[New KB Article] Why did the comments on a PR or MR not appear inline?

### PR DESCRIPTION
**Pull request by armchairlinguist**
[CSE-519](https://linear.app/semgrep/issue/CSE-519/why-were-prmr-comments-left-as-summary-comments-instead-of-inline) (internal link).

Explains the two most common reasons why Semgrep comments on a PR or MR might not appear inline.

### Please ensure

- [ ] A subject matter expert (SME) reviews the content
- [ ] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team

### CODE REVIEW

**Comment by armchairlinguist** [comment]
@chmccreery Besides a general check for whether I said anything that's false, I have two main questions for your review -

* Is the number five (for findings being posted as unanchored vs inline) correct? The internal docs say 3, but the code appeared to indicate 5.
* Is Bitbucket affected by the diff comment line limits? I had more trouble understanding the code/testing this than I did with the others.

File: docs/kb/semgrep-cloud-platform/inline-pr-comments.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import MoreHelp from "/src/components/MoreHelp"
+
+# Why did the comments on a PR or MR not appear inline?
+
+When Semgrep comments on PR or MR findings, the comments are usually posted on the line of code where the finding is identified (inline). However, there are two common reasons why comments may not appear inline.
+
+## Avoiding excessive inline comments
+
+For rules with several findings in the same PR or MR, inline comments for every individual finding occupy a significant amount of space without providing much additional information. Therefore, if the same finding occurs five or more times in a PR or MR, the related comment is posted as a summary (overall) comment.
```
Comment (chmccreery) [inline docs/kb/semgrep-cloud-platform/inline-pr-comments.md:15]:
```suggestion
For rules with several findings in the same PR or MR, inline comments for every individual finding occupy a significant amount of space without providing much additional information. Therefore, if the same finding occurs more than five times in a PR or MR, the related comment is posted as a summary (overall) comment.
```

per https://github.com/returntocorp/semgrep-app/blob/4d1d2eb1de6165f5fb926a21f4a24e8e263ed469/server/semgrep_app/cloud_platform/scm_comments/controllers/base_comments.py#L1004

**Review by chmccreery — APPROVED** [review]

File: docs/kb/semgrep-cloud-platform/inline-pr-comments.md (hunk 1 above)
Comment (armchairlinguist) [reply docs/kb/semgrep-cloud-platform/inline-pr-comments.md:15]: Ah yes thank you! I must have lost the < vs <= in memory translation.

**Comment by armchairlinguist** [comment]
TWs, if you get to this after merge, I'd be very happy for any feedback still :) Just want to get this out there.

**Pull request merged**